### DISCUSSION TITLE
[Question] What's the purpose of the "badfilter" directive, semantical-wise?

### DISCUSSION THREAD

**Discussion by MasterInQuestion** [uBlockOrigin/uBlock-issues#2183] [General] [answered]
&#xA0;&#xA0;&#xA0;&#xA0;According to the documentation [ [https://github<i>.</i>com/gorhill/uBlock/wiki/Static-filter-syntax#badfilter](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#badfilter) ], it appears to be much no more than... merely commenting?
&#xA0;&#xA0;&#xA0;&#xA0;Which I suppose is already possible with <a href="https://regex101.com/?regex=^!&flags=m">/^!/m</a> ("!" at start of line)?

**Comment by mapx-** [comment, marked as answer]
uBO is using not only its own lists.
If a wrong filter has been added to Easylist for example, you have a tool to nullify it, badfiltering it:

wrong filter:
`||example.com/player.js`

A uBO maintainer will add in this care (in unbreak list)
`||example.com/player.js$badfilter`

and the Easylist filter will cease to exist for uBO

**Reply by MasterInQuestion** [reply]
&#xA0;&#xA0;&#xA0;&#xA0;So there's some other filter implementation that does not support commenting (or does not use "!" commenting)?
&#xA0;&#xA0;&#xA0;&#xA0;Thus I assume "badfilter" directive is for compatibility ported from some other filter implementation?

**Reply by mapx-** [reply]
There is nothing about "commenting" in using `badfilter`

The wiki is very clear: **To disable an existing filter**

**Reply by ghost** [reply]
> So there's some other filter implementation that does not support commenting (or does not use "!" commenting)?     Thus I assume "badfilter" directive is for compatibility ported from some other filter implementation?

You cannot "comment" in filter list maintained by someone else and downloaded from remote location.
